packages/dds/tree/src/feature-libraries/editable-tree/editableTreeSequence.ts

```diff
@@ -0,0 +1,382 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { TypedJsonCursor } from "../../domains";
+import { ITreeCursor, ITreeSubscriptionCursor, TreeNavigationResult } from "../../forest";
+import { LocalFieldKey, NamedTreeSchema, TreeSchema, TreeSchemaIdentifier, lookupTreeSchema } from "../../schema-stored";
+import { Anchor, UpPath, Value } from "../../tree";
+import {
+    FieldlessEditableTree, getTypeSymbol, inProxyOrUnwrap, insertRootSymbol, ProxyTarget, proxyTargetSymbol,
+    UnwrappedEditableTree, valueSymbol,
+} from "./editableTree";
+import { ProxyContext } from "./editableTreeContext";
+import { AdaptingProxyHandler, getFieldSchema, isPrimitive, isPrimitiveValue } from "./utilities";
+
+/**
+ * A symbol to append a node to the sequence field in contexts where string keys are already in use for fields.
+ */
+export const appendNodeSymbol: unique symbol = Symbol("editable-tree:appendNode()");
+
+/**
+ * Unwrapped sequence field.
+ * 
+ * Limited support of unwrapped sequences under a primary key:
+ * - set value using assignment by index
+ * - append to the tail using `push()` or assignment by index equals length. This is limited to single-type sequences
+ * - append to the tail using `appendNodeSymbol`
+ * 
+ * TODO:
+ * - support generic (or implicit) sequences. This should automatically enable lazy access.
+ */
+export type UnwrappedEditableSequence = FieldlessEditableTree & readonly UnwrappedEditableTree[] & {
+    readonly [appendNodeSymbol]: (node: ITreeCursor) => void;
+};
+
+/**
+ * A Proxy target, which together with a {@link sequenceHandler} implements a basic read/write access to
+ * the sequence fields by means of the cursors.
+ */
+export class ProxyTargetSequence extends Array<ProxyTarget | ProxyTargetSequence> {
```

Comment:
Extending types and overrising their functionality can be problematic (ex: if they change which of the public API functions methods internally call it can break things, or you can accidently collide with names of private members).

Might be better to wrap an array instead. If you really want to API of an array, can `implement` array instead of extending it.

We shouldn't need to both use a proxy, and extend array. Either one should be enough to handle intercepting indexed access.